src/backend/oneapi/kernel/iir.hpp

```diff
@@ -124,10 +126,26 @@ void iir(Param<T> y, Param<T> c, Param<T> a) {
         read_accessor<T> cAcc{*c.data, h};
         read_accessor<T> aAcc{*a.data, h};
 
-        auto s_z = sycl::local_accessor<T>(MAX_A_SIZE, h);
-        auto s_a = sycl::local_accessor<T>(MAX_A_SIZE, h);
+        unsigned num_a = a.info.dims[0];
+
+        auto s_z = sycl::local_accessor<T>(num_a, h);
+        auto s_a = sycl::local_accessor<T>(num_a, h);
         auto s_y = sycl::local_accessor<T>(1, h);
 
+        size_t local_bytes_req = (num_a * 2 + 1) * sizeof(T);
+        if (local_bytes_req >
```

Comment:
can you move this to the src/backend/oneapi/level. This is not the right place to handle these types of errors. How do we handle this scenarios in the other backends?